src/thor/astar_bss.cc

```diff
@@ -29,10 +29,11 @@ constexpr uint32_t kInitialEdgeLabelCount = 500000;
 constexpr uint32_t kMaxIterationsWithoutConvergence = 200000;
 
 // Default constructor
-AStarBSSAlgorithm::AStarBSSAlgorithm(uint32_t max_reserved_labels_count)
+AStarBSSAlgorithm::AStarBSSAlgorithm(uint32_t max_reserved_labels_count, bool clear_reserved_memory)
```

Comment:
i was just talking to @genadz about this the other day. we should just pass the config ptree here and then pull the values out that we need otherwise the constructors arguments will get really long. what does everyone else think about that? the default can be an empty ptree and then we can have defaults for these values in the initializer list someting like:

```c++
AStarBSSAlgorithm::AStarBSSAlgorithm(const boost::property_tree::ptree& config = {}): 
      PathAlgorithm(), max_label_count_(std::numeric_limits<uint32_t>::max()),
      mode_(TravelMode::kDrive), travel_type_(0),
      max_reserved_labels_count_(config.get<uint32_t>("max_reserved_labels_count", kMaxReservedLabelsCountDefault),
      clear_reserved_memory_(config.get<bool>("clear_reserved_memory", false)) {
...
```

Reply:
> what does everyone else think about that?

@kevinkreiser yes, i like it too. Make it in this PR?

Reply:
> Make it in this PR?

done)